app/lib/tool/test_profile/importer.dart

```diff
@@ -0,0 +1,145 @@
+// Copyright (c) 2020, the Dart project authors.  Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+import 'dart:io';
+
+import 'package:gcloud/db.dart';
+import 'package:gcloud/service_scope.dart';
+import 'package:http/http.dart';
+import 'package:meta/meta.dart';
+import 'package:path/path.dart' as p;
+
+import 'package:client_data/package_api.dart';
+
+import '../../account/backend.dart';
+import '../../account/models.dart';
+import '../../package/backend.dart';
+import '../../publisher/models.dart';
+import '../../shared/urls.dart' as urls;
+
+import 'models.dart';
+import 'resolver.dart';
+
+/// Imports [profile] data into the Datastore.
+@visibleForTesting
+Future<void> importProfile({
+  @required TestProfile profile,
+  List<String> resolvedVersions,
+  @required String archiveCachePath,
+}) async {
+  // resolve versions if they are not yet resolved
+  if (resolvedVersions == null || resolvedVersions.isEmpty) {
+    resolvedVersions = await resolveVersions(profile);
+  }
+
+  // create users
+  for (final u in profile.users) {
+    final oauthUserId = _baseIdFromEmail(u.email);
+    final userId = _userIdFromEmail(u.email);
+    await dbService.commit(inserts: [
+      User()
+        ..id = userId
+        ..oauthUserId = oauthUserId
+        ..email = u.email
+        ..created = DateTime.now().toUtc()
+        ..isDeleted = false,
+      OAuthUserID()
+        ..id = oauthUserId
+        ..userIdKey = dbService.emptyKey.append(User, id: userId),
+    ]);
+  }
+
+  // create publishers
+  for (final p in profile.publishers) {
+    await dbService.commit(inserts: [
+      Publisher()
+        ..id = p.name
+        ..contactEmail = p.members.isEmpty ? null : p.members.first.email
+        ..websiteUrl = 'https://${p.name}/'
+        ..created = DateTime.now().toUtc()
+        ..updated = DateTime.now().toUtc()
```

Comment:
Again it would probably be a good idea to have optional support for this in `TestProfile` with a fallback to a `defaultPublishedCreated` date-time.